app/src/main/java/ml/docilealligator/infinityforreddit/customtheme/CustomTheme.java

```diff
@@ -298,4 +323,67 @@ public static CustomTheme convertSettingsItemsToCustomTheme(ArrayList<CustomThem
 
         return customTheme;
     }
+
+    private static class CustomThemeSerializer implements JsonSerializer<CustomTheme> {
+        @Override
+        public JsonElement serialize(CustomTheme src, Type typeofSrc, JsonSerializationContext context) {
+            JsonObject obj = new JsonObject();
+
+            for (Field field : src.getClass().getDeclaredFields()) {
+                try {
+                    if (field.getType() == int.class) {
+                        obj.addProperty(field.getName(), String.format("#%08X", field.getInt(src)));
+                    } else {
+                        obj.add(field.getName(), context.serialize(field.get(src)));
+                    }
+                } catch (IllegalAccessException ignored) {
+                }
+            }
+            return obj;
+        }
+    }
+
+    private static class CustomThemeDeserializer implements JsonDeserializer<CustomTheme> {
+        @Override
+        public CustomTheme deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+            CustomTheme customTheme = new CustomTheme();
+
+            JsonObject obj = json.getAsJsonObject();
+
+            for (Map.Entry<String, JsonElement> entry : obj.entrySet()) {
+
+                Field field;
+                try {
+                    field = customTheme.getClass().getDeclaredField(entry.getKey());
+                } catch (NoSuchFieldException e) {
+                    // Field not found, skip
+                    continue;
+                }
+
+                JsonElement value = entry.getValue();
+
+                try {
+                    Class<?> type = field.getType();
+                    if (int.class.equals(type)) {
+                        if (value.getAsJsonPrimitive().isString()) {
+                            // Hex color string
+                            field.set(customTheme, Color.parseColor(value.getAsString()));
```

Comment:
this will throw `IllegalArgumentException` and crash the app if string is not a valid color

Reply:
Thanks, now handled as a parse exception, which just shows the parse theme error toast.